sycl/test-e2e/KernelFusion/abort_fusion.cpp

```diff
@@ -82,11 +84,11 @@ int main() {
 
   // Scenario: Fusing two kernels with different local size should lead to
   // fusion being aborted.
-  performFusion<class Kernel1_3, class Kernel2_3>(q, range<1>{dataSize},
-                                                  range<1>{16});
-  // CHECK:      ERROR: JIT compilation for kernel fusion failed with message:
+  int num_errors = performFusion<class Kernel1_3, class Kernel2_3>(
+      q, range<1>{dataSize}, range<1>{16});
+  // CHECK: ERROR: JIT compilation for kernel fusion failed with message:
   // CHECK-NEXT: Cannot fuse kernels with different offsets or local sizes
-  // CHECK-NEXT: COMPUTATION OK
+  // CHECK: COMPUTATION OK
 
-  return 0;
+  return num_errors == 0 ? 0 : 1;
```

Comment:
I'm fine with either non-zero exit code or assert. However, using FileCheck for that purpose is unnatural and an overkill, IMO. Other tests are doing just fine using just the exit code.

Also, checking exit code allows something like

```
while ./a.out >/dev/null 2>&1 ; do true ; done
```

when working with flaky failures.

Reply:
I agree, using `FileCheck` to check for computation results is overkill, therefore, for most kernel fusion tests, we use asserts/exit code to detect failure. 

This specific test case however uses `FileCheck` for a specific reason, documented at the beginning of the file: 
> Also check that warnings are printed when SYCL_RT_WARNING_LEVEL=1.

If you prefer to also have a non-zero exit code, `assert` could be used and I'm fine with replacing `CHECK-NEXT` with `CHECK` to accommodate additional output with `ze_debug`. The remaining changes to the test-case are not really necessary IMHO.

Reply:
Do you want me to assert inside `test` or `main`? We currently only have one call to `test` but if you plan on changing that then asserting in `main` provides more output in case of early failure.

Reply:
Asserting in `performFusion` should be enough for now.